doc/source_1.tex

```diff
@@ -25,7 +25,9 @@
 &&                       && |   &\quad &&  \textit{block} 
                                                            && \textrm{block statement}\\
 &&                       && |   &\quad &&  \textit{expression} \ \textbf{\texttt{;}}
-                                                           && \textrm{expression statement} \\[1mm] 
+                                                           && \textrm{expression statement} \\
+&&                       && |   &\quad &&  \epsilon
```

Comment:
Empty statements in Javascript are represented by a single semicolon, and is currently not valid in Source. If this is to allow the 

```javascript
if (true) {
  do_something();
} else { /*nothing here*/}
```

idiom that is being used, I think block should be changed instead to allow the empty block:

```block ::= {\epsilon | statement}```

and drop the empty statement definition.

In either case, the `while` definition in https://github.com/source-academy/js-slang/blob/355564134e836d56bb6b426252e629d1fe00643d/doc/source_loops.tex#L28 should drop the redundant `undefined;` statement.

Reply:
Currently a single semicolon as a statement is not allowed (results in " Empty statements are not allowed"), but a blank statement (nothing, without semicolon) is allowed, also referred to as an empty statement in the Source specifications.

This results in "empty statement" having two definitions (one referring to ` ` while one referring to `;`), which could be confusing.

Since empty statement in JavaScript already means `;`, maybe the name of ` ` as a statement could be changed?

Or we could just change the error message currently regarding ; statements to say "; is not a valid statement".

Either way, I believe that this dual usage of the term "empty statement" should be addressed.

Reply:
I agree with your reasoning. I think we should move to JavaScript here, and allow the empty JavaScript statement
;
and disallow nothing to represent a statement.

Reply:
Yes, I'm proposing to have
;
as the empty statement, following JavaScript.